Core/Resource/SymbolJson.cs

```diff
@@ -30,218 +30,215 @@ public static void WriteObject(this JsonTextWriter writer, string name, object v
             }
         }
     }
-
-    public class SymbolJson
+    
+    public static class SymbolJson
     {
-        public JsonTextWriter Writer { get; set; }
-        public JsonTextReader Reader { get; init; }
-
         #region writing
-        public void WriteSymbol(Symbol symbol)
+        public static void WriteSymbol(Symbol symbol, JsonTextWriter writer)
         {
-            Writer.WriteStartObject();
+            writer.WriteStartObject();
 
-            Writer.WriteObject("Name", symbol.Name);
-            Writer.WriteValue("Id", symbol.Id);
-            Writer.WriteObject("Namespace", symbol.Namespace);
+            writer.WriteObject(JsonKeys.Name, symbol.Name);

```

Comment:
you'll see this JsonKeys.[whatever] everywhere in this class. It's basically paranoia but I think it makes working with Json in the future a little easier.

The idea is that 
1. now the compiler HAS TO make these keys constants, so it's basically GC-free
2. it's far more difficult to make a typo
3. it provides an easy high-level overview of the keys present in the json we are concerned with

Reply:
I'm a big fan of the nameof() idea - it'd definitely be less efficient, but at a future point we can even manage saving by reflection using that way (if that's ever actually necessary or helpful - doubt it). Though it would definitely invalidate our current files if we don't find the right cocktail for ensuring we're excluding class and namespace names. I don't think that's a tall order, though.

On your point about assumption 1, I'm just referencing the fact that string literals are not guaranteed to be compile-time constants - the compiler(s) make those decisions but that's obv a big black box to me. In the event that the compiler decides to allocate new strings for every function call instead of making them constants, that's where the GC could kick in